### DISCUSSION TITLE
Is KEYCLOAK_FRONTEND_URL environnement variable provided to docker supposed to overwrite realm's frontend url ?

### DISCUSSION THREAD

**Discussion by omasseau** [keycloak/keycloak#20868] [Q&A] [answered]
Hello,

I'm really confused about this (I'm using Keycloak 15 docker image - Yes I know its old, but I cannot upgrade right now).

Is the KEYCLOAK_FRONTEND_URL environnement variable provided to the docker container supposed to overwrite all realms' frontend url ?
This is how I understand it but I'm not sure as I found Keycloak behaviour really strange.

Let's say I defined the front end url for my master realm like this : http://localhos:8180/auth (see the missing t ?)
-> I cannot acces the console anymore as the hostname is not valid.

Now I want to force it to http://localhost:8180/auth by definining the KEYCLOAK_FRONTEND_URL variable provided to the Keycloak container in order to be able to access the console again.

If I open my browser to http://localhost:8180/auth and put my mouse over "Administration console" it looks like it is indeed using the KEYCLOAK_FRONTEND_URL variable (http://localhost:8180/auth). Great so far.

But when I click on it I get a blank page because of an ajax request is made to http://localhos:8180/auth/js/keycloak.js?version=hilhk (so it seems that here it is the frontend url defined at the realm's level that is used)

This leads me to think that KEYCLOAK_FRONTEND_URL variable is not really setting a fixed frontend url. The front end url defined at the realm's level seems to override it for some urls.

If so, how I am supposed to be able to access the console if I misstyped the frontend url while defining it throught the 'frontend url' field of the master realm ?
Is the only solution is to modify it directly in the keycloak's database ?

**Comment by jonkoops** [comment]
We're aware of this confusion, see #14666

**Reply by andre-nascimento6791** [reply]
@jonkoops , this is a bit of a new case. We plan to improve the documentation to highlight this behavior too.

Thanks.

**Comment by vmuzikar** [comment]
CC @andre-nascimento-rh @abstractj

**Reply by vmuzikar** [reply]
Actually, this question seems to be related to quite old Keycloak version that we don't support anymore. @omasseau Please consider upgrading.

**Reply by andre-nascimento6791** [reply, marked as answer]
Hello, @omasseau,

As @vmuzikar mentioned above, you're running a quite old Keycloak version, and we aren't able to provide maintenance in previous versions. You'll benefit of a lot of advantages in upgrading to the newest version.

However, I tested what you described in Keycloak 21.1.1, and I could confirm that the URL set in the `Frontend URL` field in Realm settings form (at Admin Console) is considered even over the new set of options ([`Hostname` settings](https://www.keycloak.org/server/hostname)). I believe this is the expected behavior.

Regarding your question:
`If so, how I am supposed to be able to access the console if I misstyped the frontend url while defining it throught the 'frontend url' field of the master realm? Is the only solution is to modify it directly in the keycloak's database ?`

It seems that is the solution for that situation. I needed to run a SQL Update like this: 

`UPDATE public.realm_attribute SET value = null WHERE name = 'frontendUrl' AND realm_id = 'master' `


Thanks for your efforts in using Keycloak.

**Reply by andre-nascimento6791** [reply]
We plan to improve the documentation to highlight this behavior.

Thanks.

**Reply by omasseau** [reply]
Thanks a lot for your answer 🙂

It would be great in the future to be able to override the front-end URL of the master realm through an environment variable in case one messed up the URL.
Sometimes accessing the db can be difficult (if it is in a docker container only exposing its ports to the private docker network for example) 🙂
